2294/2/analyze.rb

```diff
@@ -0,0 +1,81 @@
+require_relative './file_parser.rb'
+require 'russian_obscenity'
+require 'yaml'
+RussianObscenity.dictionary = [:default, 'dictionary.yml']
+
+# Analyze text of current artist
+# :reek:UtilityFunction
+class Analyze
+  WORD_SETTING = YAML.safe_load(File.read('pluralize_settings.yml'))
+  DEFAULT_BATTLES = 3
+  attr_reader :name, :battles
+
+  def initialize(key, value)
+    @name = key
+    @battles = value
+  end
+
+  def row
+    [name_column, battles_column, bad_words_column, bad_words_per_battle_column, words_per_round_column]
+  end
+
+  private
+
+  def text
+    @battles.join(' ')
+  end
+
+  def words
+    text.split(' ')
+  end
+
+  def battles_count
+    @battles.size
+  end
+
+  def rounds
+    battles_count * DEFAULT_BATTLES
+  end
+
+  def bad_count
```

Comment:
you should read about memoization: http://gavinmiller.io/2013/basics-of-ruby-memoization/